### PR TITLE
 Paginated, searchable people page with other enhancements

### LINKED ISSUE #1326: Paginated, searchable people page with other enhancements
As part of RFC #1320 and the project [Bring work in WorkingFamilies/Spoke fork into MoveOnOrg/Spoke](https://github.com/MoveOnOrg/Spoke/projects/14), merge the work in these WorkingFamiles/Spoke PRs:

- [101](https://github.com/WorkingFamilies/Spoke/pull/101) 
- [105](https://github.com/WorkingFamilies/Spoke/pull/105) 
- [110](https://github.com/WorkingFamilies/Spoke/pull/110) 
- [137](https://github.com/WorkingFamilies/Spoke/pull/137) 
- [140](https://github.com/WorkingFamilies/Spoke/pull/140)

### PR DESCRIPTION
**Pull request by lperson**
# Fixes #1326

This PR brings people page enhancements over from the WFP fork.

## Description

* Filter people by
  * Campaign
  * Role (texter, admin, etc.)
  * Email, first name, last name
* Sort people by
  * First name
  * Last name
  * Oldest
  * Newest
* Paginated results
* Number of people per page choices are configurable
* Default sort is configurable

This is bigger than 300 lines because of tests and also because it's a big chunk of work.

Screen Shot 2020-01-03 at 8 51 07 PM

# Checklist:

- [x] I have manually tested my changes on desktop and mobile
- [x] The test suite passes locally with my changes
- [x] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [No] ~[My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer~
- [x] I have made any necessary changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [N/A] ~My PR is labeled [WIP] if it is in progress~

### CODE REVIEW

**Comment by ibrand** [comment]
@lperson is this one out of WIP state now Larry? Should I assign a reviewer?

**Comment by lperson** [comment]
> @lperson is this one out of WIP state now Larry? Should I assign a reviewer?

No.  I'm still writing tests. 😷 🤧 🎉

**Comment by bdatkins** [comment]
Deployment is throwing an error ReferenceError: INITIAL_ROW_SIZE is not defined on https://spoke-review-lp-1326-pa-zsgokr.herokuapp.com/admin/1/campaigns

File: __test__/backend.test.js [hunk 1]
```diff
@@ -697,3 +708,80 @@ describe("Campaign", () => {
...
+          lastName: "TestTexterLast",
+          cell: "555-555-6666",
+          email: "testtexter@example.com"
+        }
+      }
+    });
+  });
+
+  it("updates the user if it is called with a userId and userData", async () => {
+    const userData = {
```
Comment (bdatkins) [inline __test__/backend.test.js:757]: Amusing test user :)

File: __test__/server/api/people.test.js [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+  let organizationId;
+  let variables;
+  let cursor;
+
+  beforeEach(async () => {
+    await setupTest();
+
+    testAdminUsers = [];
+    testAdminUsers.push(
+      await createUser({
```
Comment (bdatkins) [inline __test__/server/api/people.test.js:36]: Would there be any value in moving reused test data to a common place to reduce duplication?

File: src/server/api/user.js [hunk 3]
```diff
@@ -74,10 +80,58 @@ export function buildUserOrganizationQuery(
...
-      .leftOuterJoin("assignment", "assignment.user_id", "user.id")
-      .where({ "assignment.campaign_id": campaignId });
+  if (filterString) {
+    const filterStringWithPercents = (
+      "%" +
+      filterString +
+      "%"
+    ).toLocaleLowerCase();
+
+    if (filterBy === "FIRST_NAME") {
```
Comment (bdatkins) [inline src/server/api/user.js:90]: Suggestion to introduce constants for the filter strings

**Comment by lperson** [comment]
@bdatkins wrote: 

> Deployment is throwing an error ReferenceError: INITIAL_ROW_SIZE is not defined on https://spoke-review-lp-1326-pa-zsgokr.herokuapp.com/admin/1/campaigns

Bad merge.  I got rid of the variable in this branch.

**Review by bdatkins — APPROVED** [review]
Nice work, @lperson ! Looks good.

**Comment by bdatkins** [comment]
I verified the functionality in the Heroku environment listed above.

**Comment by lperson** [comment]
Thanks for the review and approval @bdatkins.  I'm throwing the WIP label on this @ibrand.  Please don't merge it until I've had a chance to iterate with @bdatkins's 2 review comments.

**Pull request merged**